lib/puppet/defaults.rb

```diff
@@ -727,13 +727,37 @@ def self.initialize_default_settings!(settings)
         unless [0, 'unlimited', Float::INFINITY].include?(val)
           Puppet.deprecation_warning(<<-WARNING)
 Fine grained control of environment timeouts is deprecated,
-please use `0` or `unlimited` to control default caching behavior
-and the environment-cache endpoint in Puppet Server's administrative
-API to expire the cache as needed
+please use `environment_ttl` instead.
           WARNING
         end
       end
     },
+    :environment_ttl => {
+      :default    => "0",
+      :type       => :ttl,
+      :desc       => "How long the Puppet server should cache data it loads from an
+      environment based on when it was last accessed.
```

Comment:
How long after the last access Puppet server will wait to evict an environment from the environment cache.